WebHostLib/static/assets/player-settings.js

```diff
@@ -145,8 +145,13 @@ const buildOptionsTable = (settings, romOpts = false) => {
         range.setAttribute('min', settings[setting].min);
         range.setAttribute('max', settings[setting].max);
         range.value = currentSettings[gameName][setting];
+        const settingMax = settings[setting].max;
+        const specialMax = settings[setting].specialMax;
         range.addEventListener('change', (event) => {
-          document.getElementById(`${setting}-value`).innerText = event.target.value;
+          const v = event.target.value;
```

Comment:
There is no need to assign this to memory, as it is already in memory.

Reply:
To my knowledge, accessing the property of an object already in memory is not slower than assigning a new scalar. Using `event.target.value` should be both faster and use less memory. If you have documentation which shows otherwise, I will consider it.

If you perform some calculation or operation on data which should be referenced multiple times, that is worth assigning new memory for.

As for shorter expressions, that depends on the person reading the code. As the senior developer at my company, with ten years of professional experience, it has become my opinion that shortcut variables and short lines of code with the sole purpose of being short are undesirable and cause longer code analysis times (takes longer to read). I can provide further explanation and examples of this if you'd like.

Reply:
I am interested in further explanation and examples.

> with the sole purpose of being short

The purpose is not to be short. The purpose is to be easier to understand.
I agree that the single letter variable name was a bad choice, but it should still be in separate variable, just with a better name.

It only takes longer to read if you don't care about understanding the code.
It simplifies things to separate the ideas of the value and where the value came from.
`event.target.value` contains a lot of information about where the value came from that is not relevant where the value is used.
That information would be a distraction that makes the expression more difficult to understand.
`event` isn't relevant when we use the value. `target` isn't relevant when we use the value.

It also makes it easier to debug and maintain if there are more lines for breakpoints, and more places to insert debug messages, or insert the new feature that we're implementing.

Using `event.target.value` is not faster and I doubt it uses less memory.

The interpreters are optimized for this because many people recognize the value in writing code like this.

I've done my own tests to make sure this
```
        const x = obj.x;
        const y = x[0];
        const z = y.z;
        return z;
```
is just as fast as this
```
        return obj.x[0].z;
```

Reply:
> `event.target.value` contains a lot of information about where the value came from that is not relevant where the value is used.

A value and its source are never more relevant than where they are being used. I don't understand this logic.

> That information would be a distraction that makes the expression more difficult to understand.

This argument would hold more weight in a discussion about literature, I think. If I was reading through a section of code, saw an assignment, and wanted to know where that assignment was made, I would need to look up the value's source. Using `event.target.value` in this case would prevent me from needing to look that information up.

In any case, as I have addressed, there is actually a computationally beneficial reason to use an assignment here. You're wanting to make a comparison, and should be using `===` to do so. To accomplish this, `parseInt` is required to ensure type consistency. Therefore, assigning the result of `parseInt(event.target.value, 10)` to your new `const` is preferable, and prevents repeat computation.

Reply:
> A value and its source are never more relevant than where they are being used.

`event` isn't relevant when we use the value. `target` isn't relevant when we use the value.

You're claiming that they're relevant. How are they relevant?

You're presenting a hypothetical situation where you "want to know where that assignment was made".
That's begging the question. For the understanding of that expression, "where that assignment was made" is not relevant.